repos/system_upgrade/common/actors/cloud/checkgrubenvtofile/tests/test_checkgrubenvtofile.py

```diff
@@ -0,0 +1,58 @@
+import pytest
+
+from leapp import reporting
+from leapp.libraries.actor import checkgrubenvtofile
+from leapp.libraries.common.testutils import create_report_mocked, CurrentActorMocked, produce_mocked
+from leapp.libraries.stdlib import api
+from leapp.models import FirmwareFacts, HybridImage
+
+BIOS_FIRMWARE = FirmwareFacts(firmware='bios')
+EFI_FIRMWARE = FirmwareFacts(firmware='efi')
+
+
+@pytest.mark.parametrize('is_symlink', [True, False])
+@pytest.mark.parametrize('realpath_match', [True, False])
+def test_is_grubenv_symlink_to_efi(monkeypatch, tmpdir, is_symlink, realpath_match):
+    grubenv_efi = tmpdir.join('grubenv_efi')
+    grubenv_efi.write('grubenv')
+    grubenv_efi.write('nope')
+    grubenv_boot = tmpdir.join('grubenv_boot')
+
+    grubenv_efi_false = tmpdir.join('grubenv_efi_false')
```

Comment:
Instead of relying on filesystem changes, I would suggest mocking the `os` functions used by the actor. I don't like using filesystem because filesystem is not in our exclusive control, in general.